### ISSUE TITLE
SessionState and FrameNavigationService

### ISSUE THREAD

**Issue by MichaelPonti** [PrismLibrary/Prism#505]
Hi, I have been a happy user of the prism library in wpf and recently started using it on universal apps. I was hoping to make use of the session state restore, but what I found is that it wasn’t working as expected when using a hamburger style ui. Each time I clicked on a button in the menu, even if I had already visited the page, it would treat it as a new view and stuff a new sessionstate on the stack.

I eventually traced it down into the FrameNavigationService to a couple of private methods that were generating new keys on navigation. I was just going to make my own class to handle the session state keys by viewmodel class name but thought it might be something other people would want in the UWP libraries as well.

In a nutshell, I think most of the functionality of the FrameNavigationService class can be dropped down into a base class and then have the current FrameNavigationService derive from it and continue on with generating its stack of session states. A hamburger menu style navigation service could also make use of the base class but generate viewmodel keys based on the viewmodel name instead.
Is this something you would be interested in or am I just completely missing something on how to use the FrameNavigationService …

**Comment by bartlannoeye** [comment]
Navigation in UWP is seen as forward navigation to new pages. I do understand what you'd like to see as behavior, but I'm not sure if this is something we'd like to bring in. Has to be discussed with @brianlagunas and @briannoyes.

**Comment by MichaelPonti** [comment]
You guys are in charge :) 

If it is not consistent with your direction, it would be very easy for me to just make a new class in my app project and add it in during app creation so I wouldn't be maintaining a fork of the prism library.

**Comment by brianlagunas** [comment]
The session state service is being removed from Prism for UWP in the up-coming rewrite.
